src/resultsdisassemblypage.cpp

```diff
@@ -0,0 +1,159 @@
+#include "resultsdisassemblypage.h"
+#include "ui_resultsdisassemblypage.h"
+
+#include <QMenu>
+#include <QFileDialog>
+#include <QFile>
+#include <QTextStream>
+#include <QMessageBox>
+#include <QString>
+#include <QListWidgetItem>
+#include <QProcess>
+#include <QDebug>
+#include <QDir>
+#include <QDirIterator>
+
+#include <KRecursiveFilterProxyModel>
+
+#include "parsers/perf/perfparser.h"
+#include "resultsutil.h"
+
+#include "models/costdelegate.h"
+#include "models/hashmodel.h"
+#include "models/topproxy.h"
+#include "models/treemodel.h"
+
+#include <QStandardItemModel>
+
+ResultsDisassemblyPage::ResultsDisassemblyPage(FilterAndZoomStack *filterStack, PerfParser *parser, QWidget *parent)
+        : QWidget(parent), ui(new Ui::ResultsDisassemblyPage) {
+    ui->setupUi(this);
+    ui->splitter->setStretchFactor(0, 1);
+    ui->splitter->setStretchFactor(1, 1);
+    ui->splitter->setStretchFactor(2, 8);
+}
+
+ResultsDisassemblyPage::~ResultsDisassemblyPage() = default;
+
+/**
+ *  Clear
+ */
+void ResultsDisassemblyPage::clear() {
+    if (ui->asmView->model() == nullptr) return;
+
+    int rowCount = ui->asmView->model()->rowCount();
+    if (rowCount > 0) {
+        ui->asmView->model()->removeRows(0, rowCount, QModelIndex());
+    }
+}
+
+/**
+ *  Produce and show disassembly with 'objdump'
+ */
+void ResultsDisassemblyPage::showDisassembly() {
+    showDisassemblyByAddressRange();
+}
+
+/**
+ *  Produce disassembler with 'objdump' by addresses range and output to Disassembly tab
+ */
+void ResultsDisassemblyPage::showDisassemblyByAddressRange() {
+    // Show empty tab when selected symbol is not valid
+    if (m_curSymbol.symbol.isEmpty()) {
+        clear();
+        return;
+    }
+
+    // Call objdump with arguments: addresses range and binary file
+    QString processName =
+            m_objdump + QLatin1String(" -d --start-address=0x") + QString::number(m_curSymbol.relAddr, 16) +
+            QLatin1String(" --stop-address=0x") +
+            QString::number(m_curSymbol.relAddr + m_curSymbol.size, 16) +
+            QLatin1String(" ") + m_curSymbol.path;
+
+    showDisassembly(processName);
+}
+
+/**
+ *  Produce disassembler with 'objdump' and output to Disassembly tab
+ */
+void ResultsDisassemblyPage::showDisassembly(QString processName) {
+    QTemporaryFile m_tmpFile;
```

Comment:
ideally all of this code should be rewritten to be async using signals/slots, instead of blocking the UI while we are waiting for objdump to finish

Reply:
i.e. show a busy indicator while we are waiting for the data to come in, but otherwise don't block the UI

Reply:
furthermore, please move the logic to run and parse objdump into a separate class that can be tested separately from the UI

Reply:
> i.e. show a busy indicator while we are waiting for the data to come in, but otherwise don't block the UI

Can this be refactored in this async way after all patches are merged. It is extremely complicated for me to sync with the changes in other commits when I change in a particular commit. Do you insist such async refactoring or it can be postponed?

Reply:
ok, that can be postponed, but it is very important to have

Reply:
please add at least a TODO that this needs to be extracted into a standalone function, covered by unit tests and made async

Reply:
Ok, I will add